### PR TITLE
Add toBeFinite

### PR DESCRIPTION
**Pull request by tophernuts**
Fixes https://github.com/mattphillips/jest-extended/issues/28

### What
Adds the `toBeFinite` matcher

### Why
https://github.com/mattphillips/jest-extended/issues/28

### Housekeeping

- [x] Unit tests
- [x] Documentation is up to date
- [x] No additional lint warnings
- [x] Add yourself to contributors list (`yarn contributor`)

### CODE REVIEW

File: src/matchers/toBeFinite/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  matcherHint('.not.toBeFinite', 'received', '') +
+  '\n\n' +
+  'Expected value to not be finite received:\n' +
+  `  ${printReceived(received)}`;
+
+const failMessage = received => () =>
+  matcherHint('.toBeFinite', 'received', '') +
+  '\n\n' +
+  'Expected value to be finite:\n' +
+  `  ${printExpected(1)}\n` +
```
Comment (mattphillips) [inline src/matchers/toBeFinite/index.js]: I'm not sure this print expected 1 is needed

**Comment by mattphillips** [comment]
All looks spot on other than the one tiny comment, good job :smile:

**Comment by tophernuts** [comment]
Hey @mattphillips just updated that, need me to rebase/squash these commits at all?

**Comment by mattphillips** [comment]
Hey @tophernuts na just need you to update the snapshots by running `yarn test --u` that should fix the build :smile:

**Comment by mattphillips** [comment]
Nice one @tophernuts thanks for your help 😄

**Pull request merged**